server/src/main/java/org/apache/cassandra/sidecar/tasks/KeyStoreCheckPeriodicTask.java

```diff
@@ -41,25 +41,49 @@ public class KeyStoreCheckPeriodicTask implements PeriodicTask
     private static final Logger LOGGER = LoggerFactory.getLogger(KeyStoreCheckPeriodicTask.class);
 
     private final Vertx vertx;
-    private final Provider<Server> server;
-    private final SslConfiguration configuration;
+    private final SslConfiguration sslConfiguration;
+    private final Function<Long, Future<Boolean>> updateSSLOptionsFunction;
     private long lastModifiedTime = 0; // records the last modified timestamp
+    private final String taskName;
 
-    public KeyStoreCheckPeriodicTask(Vertx vertx, Provider<Server> server, SidecarConfiguration configuration)
+    public KeyStoreCheckPeriodicTask(Vertx vertx,
```

Comment:
should we make this protected instead?
```suggestion
    protected KeyStoreCheckPeriodicTask(Vertx vertx,
```